src/server/server_state.cc

```diff
@@ -71,6 +71,15 @@ void ServerState::Destroy() {
   state_ = nullptr;
 }
 
+uint64_t ServerState::GetCachedUsedMemory(uint64_t now_ns) {
+  uint64_t kCacheEveryNs = 1000;
+  if (now_ns > last_chached_used_current_ + kCacheEveryNs) {
+    last_chached_used_current_ = now_ns;
+    used_mem_ = used_mem_current;
```

Comment:
Maybe use `used_mem_current.load(...)` to be clear that you're caching an atomic var? It was an odd method to read before I realized that :)